docs/tutorial/t-manage-passwords.md

```diff
@@ -0,0 +1,153 @@
+  security.protocol=SASL_PLAINTEXT
+  bootstrap.servers=kafka-k8s-2.kafka-k8s-endpoints:9092,kafka-k8s-0.kafka-k8s-endpoints:9092,kafka-k8s-1.kafka-k8s-endpoints:9092
+password: 0FIQ5QxSaNfl1bXHtV5dyttb21Nbzmpp
+username: admin
+```
+The admin password is under the result: `password`.
+
+### Rotate the admin password
+You can change the admin password to a new random password by entering:
+```shell
+juju run kafka-k8s/leader set-password username=admin
+```
+Running the command should output:
+```shell
+Running operation 14 with 1 task
+  - task 15 on unit-kafka-k8s-2
+
+Waiting for task 15...
+admin-password: Fz6wPkfGtgjnQ3PCJuZnzJESudZkAvrV
+```
+The admin password is under the result: `admin-password`. It should be different from your previous password.
+
+*Note when you change the admin password you will also need to update the admin password the in Kafka connection parameters; as the old password will no longer be valid.*
+
+### Set the admin password
+You can change the admin password to a specific password by entering:
+```shell
+juju run kafka-k8s/leader set-password username=admin password=my-new-password
+```
+Running the command should output:
+```shell
+Running operation 18 with 1 task
+  - task 19 on unit-kafka-k8s-2
+
+Waiting for task 19...
+admin-password: my-new-password
+```
+The admin password under the result: `admin-password` should match the password we provided as the action argument. 
+
+*Note that when you change the admin password you will also need to update the admin password in the Kafka connection parameters, as the old password will no longer be valid.*
+
+
+### Kafka Users
+
+As mentioned in the previous section of the Tutorial, the recommended way to create and manage users is by means of the data-integrator charm. 
+This will allow us to encode users directly in the Juju model, and - as shown in the following - to rotate user credentials rotations with and without application downtime.   
+
+### Retrieve the password
+
+Similarly to the Kafka application, also the `data-integrator` exposes an action to retrieve the credentials, e.g. 
+```shell
+juju run data-integrator/leader get-credentials
+```
+Running the command should output:
+```shell 
+Running operation 22 with 1 task
+  - task 23 on unit-data-integrator-0
+
+Waiting for task 23...
+kafka:
+  endpoints: kafka-k8s-2.kafka-k8s-endpoints:9092,kafka-k8s-1.kafka-k8s-endpoints:9092,kafka-k8s-0.kafka-k8s-endpoints:9092
+  password: S4IeRaYaiiq0tsM7m2UZuP2mSI573IGV
+  tls: disabled
+  topic: test-topic
+  username: relation-6
+  zookeeper-uris: zookeeper-k8s-0.zookeeper-k8s-endpoints:2181,zookeeper-k8s-1.zookeeper-k8s-endpoints:2181,zookeeper-k8s-2.zookeeper-k8s-endpoints:2181/kafka-k8s
+ok: "True"
+```
+
+As before, the admin password is under the result: `password`.
+
+### Rotate the password
+
+#### With application downtime
```

Comment:
"With application downtime / without application downtime." My opinion on this: The purpose of the tutorial is to be as direct as possible, I think showcasing these two options shouldn't be part of the tutorial. Only chose one of them. Probably the other one is better suited to a "how-to"

Reply:
Yes, indeed. I fully agree. I'll revise the text accordingly.